src/devices/Display/Matrix16x8.cs

```diff
@@ -0,0 +1,203 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.I2c;
+
+namespace Iot.Device.Display
+{
+    /// <summary>
+    /// Represents an 8x16 LED matrix using the HT16K33 LED Matrix driver.
+    /// </summary>
+    // Product: https://www.adafruit.com/product/2042
+    public class Matrix16x8 : Ht16k33
+    {
+        private readonly byte[] _displayBuffer = new byte[17];
+
+        /// <summary>
+        /// Initialize Matrix display
+        /// </summary>
+        /// <param name="i2cDevice">The <see cref="I2cDevice"/> to create with.</param>
+        public Matrix16x8(I2cDevice i2cDevice)
+            : base(i2cDevice)
+        {
+        }
+
+        /// <summary>
+        /// Indexer for updating matrix.
+        /// </summary>
+        public int this[int x, int y]
+        {
+            set
+            {
+                UpdateBuffer(x, y, value);
+
+                if (BufferingEnabled)
+                {
+                    Flush();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Enable all LEDs.
+        /// </summary>
+        public void Fill()
+        {
+            Span<byte> displayBuffer = _displayBuffer;
+            displayBuffer.Fill(0xFF);
+            displayBuffer[0] = 0x00;
+
+            if (BufferingEnabled)
+            {
+                _i2cDevice.Write(_displayBuffer);
+            }
+        }
+
+        /// <inheritdoc/>
+        public override void Clear()
+        {
+            _displayBuffer.AsSpan().Clear();
+
+            if (BufferingEnabled)
+            {
+                _i2cDevice.Write(_displayBuffer);
+            }
+        }
+
+        /// <inheritdoc/>
+        public override void Flush() => _i2cDevice.Write(_displayBuffer);
+
+        /// <inheritdoc/>
+        public override void Write(ReadOnlySpan<byte> data, int startAddress = 0)
+        {
+            // startAddress is intended to mean which matrix to write to
+            // only 8 bytes are supported at a time
+            int rowIndex = 0;
+            foreach (byte b in data)
+            {
+                int column = (rowIndex * 2) + 1 + startAddress;
+                _displayBuffer[column] = b;
+                rowIndex++;
+            }
+
+            if (BufferingEnabled)
```

Comment:
as you have those 4 lines in many places, does it make sense to replace it by a simple function?
somthing like this?
```csharp
private void FlushIfNeed() => BufferEnabled ? Flush(): return;
```